dfsclient/src/main/java/org/hpccsystems/dfs/client/BinaryRecordWriter.java

```diff
@@ -475,10 +475,13 @@ else if (fd.getDataLen() == 8)
             case STRING:
             {
                 String value = fieldValue != null ? (String) fieldValue : "";
-                int eosIdx = value.indexOf('\0');
-                if (eosIdx > -1)
+                if (fd.getFieldType() == FieldType.VAR_STRING)
```

Comment:
minor: why not introduce nullByte (from L:614) earlier or as static final and use that throughout instead of the literal '\0'